### PR TITLE
!!! BUGFIX: Remove unused MD5 in ResourceMetaDataInterface

### PR DESCRIPTION
**Pull request by rkrahnen**
Removes the `md5` property from persistent resources that is obsolete since
we introduced a `sha1` field.

Note:

This change is marked breaking because it removes the two API methods
`PersistentResource::getMd5()` and `PersistentResource::setMd5()`.
In practice this should never be an issue really.

Instead of `PersistentResource::getMd5()` one could use `PersistentResource::getSha1()`
in order to retrieve a unique hash for a file.
In case a MD5 hash is required, the following one can do `md5(stream_get_contents($resource->getStream()))`

Fixes: #2112

### CODE REVIEW

**Comment by rkrahnen** [comment]
Okay, I think the test fails because of the remaining `$md5` property and the column not being nullable. So I have to add the migration first, but therefor I need some information.

The up migration is easy, I just drop the column in the right resource table after checking which table exists like `Version20180622074421` for example.
But what about the down migration? I could just add the column again, but it would be empty for existing entries. Would that be fine for you as this is something for the next major, or do you have any other idea?

**Comment by sorenmalling** [comment]
Maybe this can help


https://github.com/neos/flow/blob/master/Migrations/Mysql/Version20141015125841.php

There is a hook after up and down you can use to insert values

**Comment by sorenmalling** [comment]
@kitsunet @bwaidelich Main thing to review is the way of the postDown - the way resources is retrieved, we get's a instance of the resourceRepository. Is it safe to do so at this point, in this way?

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (bwaidelich) [inline Neos.Flow/Migrations/Mysql/Version20200908155620.php:55]: This should be a temporary path IMO, see `Environment->getPathToTemporaryDirectory()`

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (bwaidelich) [inline Neos.Flow/Migrations/Mysql/Version20200908155620.php:61]: not important, but this would make it easier to follow IMO:
```php
if (!is_resource($stream)) {
  continue;
}
```

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (bwaidelich) [inline Neos.Flow/Migrations/Mysql/Version20200908155620.php:4]:
```suggestion
use Doctrine\Migrations\AbstractMigration
```
image

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (bwaidelich) [inline Neos.Flow/Migrations/Mysql/Version20200908155620.php:71]: Do we really need a file here?
This seems to work just fine:

```php
/* @var PersistentResource $resource */
foreach ($resourceRepository->iterate($iterator) as $resource) {
    if (!\is_resource($resource->getStream())) {
        continue;
    }
    $this->connection->executeUpdate(
        'UPDATE neos_flow_resourcemanagement_persistentresource SET md5 = ? WHERE persistence_object_identifier = ?',
        [md5(stream_get_contents($resource->getStream())), $persistenceManager->getIdentifierByObject($resource)]
    );
}
```

**Review by bwaidelich — CHANGES_REQUESTED** [review]
Thanks for this!
I added a few suggestions. Apart from those this change should be marked breaking since it (strictly speaking) removes a method that was marked `@api`

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (rkrahnen) [reply Neos.Flow/Migrations/Mysql/Version20200908155620.php]: Removed file.

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (rkrahnen) [reply Neos.Flow/Migrations/Mysql/Version20200908155620.php]: Done.

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (rkrahnen) [reply Neos.Flow/Migrations/Mysql/Version20200908155620.php]: Done.

File: Neos.Flow/Migrations/Mysql/Version20200908155620.php
Comment (rkrahnen) [reply Neos.Flow/Migrations/Mysql/Version20200908155620.php]: Good catch, changed it.

**Comment by rkrahnen** [comment]
@bwaidelich I agree with you about the change being breaking. If there is anything I can do, let me know.

**Review by bwaidelich — APPROVED** [review]
Thanks for the update.
I like the current solution but I wonder if we should change the target branch to `master` since it is potentially a breaking change

PS: I already took the freedom to adjust the PR description to mark it breaking

**Comment by sorenmalling** [comment]
>  I like the current solution but I wonder if we should change the target branch to `master` since it is potentially a breaking change

I support that idea 👍

**Comment by rkrahnen** [comment]
Are we waiting for @kitsunet to approve this as well?

**Comment by bwaidelich** [comment]
> Are we waiting for @kitsunet to approve this as well?

No, not really. But the CI tests fail and IMO you should switch the base branch to `master` (maybe that resolves the broken CI)

You can switch the branch by editing the PR:
image

**Review by sorenmalling — APPROVED** [review]

**Comment by sorenmalling** [comment]
Approved, if branch is changed to "master" and we can get the tests running 👍

**Review by kdambekalns — APPROVED** [review]
Looks good! I restarted the TravisCI jobs, seem to pass now.

**Comment by rkrahnen** [comment]
@bwaidelich : Are you taking care of marking the required parts as deprecated in 6.3?

**Comment by bwaidelich** [comment]
> Are you taking care of marking the required parts as deprecated in 6.3?

Which parts?

**Comment by bwaidelich** [comment]
Ah, I see what you mean, marking `get/setMd5()` deprecated for 6.3.x!? I have too much on my list right now, but I can take care next week if nobody else steps up

**Comment by bwaidelich** [comment]
Ah, heck.. => #2115

**Comment by rkrahnen** [comment]
Awesome, thanks!

**Pull request merged**
